aesara/link/jax/jax_dispatch.py

```diff
@@ -1104,9 +1104,14 @@ def batched_dot(a, b):
 @jax_funcify.register(RandomVariable)
 def jax_funcify_RandomVariable(op):
     name = op.name
-
+    
+    if not hasattr(jax.random, name):
+        raise NotImplementedError(f"No JAX conversion for the given distribution: {name}")
+    
     def random_variable(rng, size, dtype, *args):
-        smpl_value = getattr(jax.random, name)(rng, size)
+        dtype = jnp.dtype(dtype)
+        data = getattr(jax.random, name)(key=rng, shape=size)
+        smpl_value = jnp.array(data, dtype=dtype)
         return (dtype, smpl_value)
```

Comment:
This should really return the `rng` state **after** sampling&mdash;as the first element in the tuple, instead of `dtype`.  I don't know if that's available from JAX, but that's what the constructed graph expects.  

Problems will arise when that `dtype` value is mistakenly used as an RNG by other `Op`s.  Those values are only ever used by other `RandomVariable` `Op`s, and, if we return the aforementoined post-sampling state values (in the form expected by JAX samplers), things should work fine.

Reply:
>  I don't know if that's available from JAX.

As far as I know JAX doesn't provide that functionality. The best we can do is provide the distribution the same `PRNGKey` (which is basically a `DeviceArray`) cause all the processing and state handling takes place inside the functions.   

The `jax.random` [documentations](https://jax.readthedocs.io/en/latest/jax.random.html) also state:

```Note that using a key does not modify it, so reusing the same key will lead to the same result``` 

I think the best course of action here will be returning `rng` variable instead of `dtype`

Reply:
It looks like the answer is to use `jax.random.split`.

Reply:
Ah, I misssed that. Basically the key itself doesn't change upon sampling. We have to change it manually by calling `jax.random.split`